include/cuco/detail/open_addressing_impl.cuh

```diff
@@ -125,11 +125,33 @@ class open_addressing_impl {
                                  Allocator const& alloc,
                                  cuda_stream_ref stream) noexcept
     : empty_key_sentinel_{empty_key_sentinel},
+      empty_slot_sentinel_{empty_slot_sentinel},
       predicate_{pred},
       probing_scheme_{probing_scheme},
       storage_{make_valid_extent<cg_size, window_size>(capacity), alloc}
   {
-    storage_.initialize(empty_slot_sentinel, stream);
+    this->clear(stream);
```

Comment:
Now that we have both `clear` and `clear_async` I'm raising the question which version the ctor should call?

Reply:
`clear_async`

Reply:
Could this scenario get us into trouble?
```cpp
cudaStream_t stream1, stream2;
// intialize stream1 and stream2

cuco::experimental::static_set<int> set(....., stream1);
set.insert_async(..., stream2); // ctor might not have finished yet
```

Reply:
worth adding a note in the ctor doc.

Reply:
Addressed in [0d00e01](https://github.com/NVIDIA/cuCollections/pull/333/commits/0d00e014801ef1bc1ced305bcd45afb9136b3edc)